routers/web/home.go

```diff
@@ -59,6 +66,55 @@ func Home(ctx *context.Context) {
 	ctx.HTML(http.StatusOK, tplHome)
 }
 
+// Home renders main sitemap
+func HomeSitemap(ctx *context.Context) {
+	m := sitemap.NewSitemapIndex()
+	if !setting.Service.Explore.DisableUsersPage {
+		_, count, err := user_model.SearchUsers(&user_model.SearchUserOptions{
+			Actor:       ctx.User,
+			Type:        user_model.UserTypeIndividual,
+			ListOptions: db.ListOptions{PageSize: 1},
+			IsActive:    util.OptionalBoolTrue,
+			Visible:     []structs.VisibleType{structs.VisibleTypePublic},
+		})
+		if err != nil {
+			ctx.ServerError("SearchUsers", err)
+			return
+		}
+		idx := 1
+		for i := int64(0); i < count; i += 50000 {
```

Comment:
can this be abused to DOS gitea?

Reply:
I had the same doubt when setting this initially, but I was also afraid of running out of subsitemaps due to a low paging size.
Then I realized that with the default explore page size of 20, even 100k repos + 100k users would only take up 10k subsitemaps, well below the limit of 50k.
Reverted to the default explore page size.